src/megaapi_impl.cpp

```diff
@@ -2773,9 +2778,12 @@ void MegaTransferPrivate::setLastBytes(char *lastBytes)
     this->lastBytes = lastBytes;
 }
 
-void MegaTransferPrivate::setLastError(MegaError e)
+void MegaTransferPrivate::setLastError(const MegaError *e)
 {
-    this->lastError = e;
+    if (lastError)
+        delete lastError;
```

Comment:
using unique_ptr would avoid the need for this code

Reply:
Agree

Reply:
Solved